src/mscorlib/shared/System/Globalization/TextInfo.cs

```diff
@@ -91,28 +91,6 @@ internal static int GetHashCodeOrdinalIgnoreCase(string s)
             return Invariant.GetCaseInsensitiveHashCode(s);
         }
 
-        // Currently we don't have native functions to do this, so we do it the hard way
-        internal static int IndexOfStringOrdinalIgnoreCase(string source, string value, int startIndex, int count)
-        {
-            if (count > source.Length || count < 0 || startIndex < 0 || startIndex > source.Length - count)
```

Comment:
I believe this condition isn't handled in CompareInfo.IndexOfOrdinal or in subsequent calls.

Reply:
same applies to LastIndexOfString*

Reply:
These conditions are covered in `src/mscorlib/shared/System/String.Searching.cs` that is the only place where this method was called from.